torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -246,8 +232,15 @@ void insertQuantizationOps(
   }
   Value* original_val = observer->input(1);
   Node *quant, *choose_qparams, *dequant;
+  bool has_dequant = true;
   if (quant_type == QuantType::DYNAMIC && isNoopObserver(observer->input(0))) {
-    dequant = insertFP16CastOps(g, observer_out);
+    // We don't need to insert cast operators for activation tensors for fp16
+    // quant.
```

Comment:
can we filter this in a different place? e.g. we don't insert observer for activation tensors?

Reply:
Is it possible for user to not specify activation observer in the qconfig? Does the prepare_jit pass ensure observers aren't inserted in that case for activation tensors?

Reply:
I mean doing checks like this: https://codebrowser.bddppq.com/pytorch/pytorch/torch/csrc/jit/passes/quantization/insert_observers.cpp.html#1187

Reply:
That may have issues since for FP16 quant we don't specify dtype anywhere in the qconfig. We set quant type to dynamic so there is no way to distinguish int8 dynamic quant from fp16 dynamic quant. Hence was wondering if not specifying any activation observer (since we don't want it observed) would work here.

Reply:
Looks like right now we are checking noop observer to do fp16 quantization, this sounds like a hack, can we expose fp16 as an argument to the API?

Reply:
i feel it makes more sense to expose this in the top level API, why don't we do that?